src/VisualStudio/Core/Def/Packaging/PackageInstallerServiceFactory.cs

```diff
@@ -136,6 +162,79 @@ private void OnSourceProviderSourcesChanged(object sender, EventArgs e)
             PackageSourcesChanged?.Invoke(this, EventArgs.Empty);
         }
 
+        public bool IsCompatible(
+            Workspace workspace,
+            ProjectId projectId,
+            IReadOnlyList<string> packageTargetFrameworkMonikers)
+        {
+            this.AssertIsForeground();
+
+            if (workspace == _workspace && _workspace != null)
+            {
+                return IsCompatible(projectId, packageTargetFrameworkMonikers);
+            }
+
+            return false;
+        }
+
+        private bool IsCompatible(ProjectId projectId, IReadOnlyList<string> packageTargetFrameworkMonikers)
+        {
+            this.AssertIsForeground();
+
+            if (packageTargetFrameworkMonikers.Count == 0)
+            {
+                // We haven't lit up target framework monikers in our DB yet.  Just assume this is
+                // ok for now.
+                return true;
+            }
+
+            if (_frameworkCompatibilityOpt == null)
+            {
+                // We don't have access to nuget's compatibility checker yet.  Just assume this is
+                // ok for now.
+                return true;
+            }
+
+            var getNearestMethod = _frameworkCompatibilityOpt.GetType().GetMethod("GetNearest", BindingFlags.Instance | BindingFlags.Public);
+            if (getNearestMethod == null)
+            {
+                // The compatibility type doesn't have hte method we expect.  Just assume this is
+                // ok for now.
+                return true;
+            }
+
+            IVsHierarchy hierarchy;
+            string projectTargetMoniker;
+            if (_workspace.TryGetHierarchy(projectId, out hierarchy) &&
+                hierarchy.TryGetProperty((__VSHPROPID)__VSHPROPID4.VSHPROPID_TargetFrameworkMoniker, out projectTargetMoniker) &&
+                !string.IsNullOrWhiteSpace(projectTargetMoniker))
+            {
+                try
+                {
+                    var projectFrameworkName = new FrameworkName(projectTargetMoniker);
+                    var packageFrameworkNames = packageTargetFrameworkMonikers.Select(m => new FrameworkName(m)).ToArray();
+
+                    try
+                    {
+                        // GetNearest has the signature:
+                        // public FrameworkName GetNearest(FrameworkName targetFramework, IEnumerable<FrameworkName> frameworks)
+                        var nearestName = getNearestMethod.Invoke(_frameworkCompatibilityOpt, new object[] { projectFrameworkName, packageFrameworkNames });
```

Comment:
This is where we need the projectmodel library to get merged into one, and then expose pieces it through an IVs api. Perhaps we should add another one here for get TFM for project?

Note that even more interesting here - You might have more than one TFM per project (even in csproj + project.json if the user explicitly added it)